marconi-core/src/Marconi/Core/Experiment/Indexer/SQLiteAggregateQuery.hs

```diff
@@ -0,0 +1,93 @@
+{-# LANGUAGE ConstraintKinds #-}
+{-# LANGUAGE TemplateHaskell #-}
+
+{- |
+    A @SQLiteAggregateQuery@ provides a way to query the content of several @SQLiteIndexer@s
+    as if they were a single indexer.
+
+    Note that @SQLiteAggregateQuery doesn't implement 'IsIndex': it won't index anything,
+    and won't pass any information to the indexers it relies on.
+    It's sole purpose is to provide a unified access for queries.
+-}
+module Marconi.Core.Experiment.Indexer.SQLiteAggregateQuery (
+  mkSQLiteAggregateQuery,
+  SQLiteSourceProvider (..),
+  IsSourceProvider,
+  SQLiteAggregateQuery (..),
+  aggregateHandle,
+  HasDatabasePath (..),
+) where
+
+import Control.Lens ((^.))
+import Control.Lens qualified as Lens
+import Control.Monad (void)
+import Control.Monad.Cont (MonadIO (liftIO))
+import Data.Map (Map)
+import Data.Map qualified as Map
+import Database.SQLite.Simple (NamedParam ((:=)))
+import Database.SQLite.Simple qualified as SQL
+
+import Control.Concurrent qualified as Con
+import Marconi.Core.Experiment.Class (Closeable (close), IsSync (lastSyncPoint))
+import Marconi.Core.Experiment.Indexer.SQLiteIndexer (SQLiteIndexer)
+import Marconi.Core.Experiment.Indexer.SQLiteIndexer qualified as SQLite
+import Marconi.Core.Experiment.Type (Point)
+
+-- | A class for indexer that has access to a SQLite database and know the path to this database
+class HasDatabasePath indexer where
+  -- | Retrieve the database path from the indexer
+  getDatabasePath :: indexer event -> FilePath
+
+instance HasDatabasePath SQLiteIndexer where
+  getDatabasePath = Lens.view SQLite.databasePath
+
+type IsSourceProvider m event indexer = (IsSync m event indexer, HasDatabasePath indexer)
+
+{- | A wrapper that provides a @SQLiteAggregateQuery@ access to
+one of the indexers it requires to answer its query
+-}
+data SQLiteSourceProvider m point
```

Comment:
Add comment that it's used for allowing heterogeous list in SQLiteAggregateQuery